toffy/json_utils.py

```diff
@@ -147,3 +155,35 @@ def split_run_file(run_dir, run_file_name, file_split: list):
         save_path = os.path.join(run_dir, run_file_name.split('.json')[0] + '_part'
                                  + str(i+1) + '.json')
         write_json_file(save_path, json_i, encoding='utf-8')
+
+
+def check_for_empty_files(bin_file_dir, return_json_names=False, warn=True):
+    """ Check for any empty json files and warn the user
+    Args:
+        bin_file_dir (str): directory containing the bin and json files
+        return_json_names (bool): whether to return a list of fovs with empty json files
+        warn (bool): whether to print a warning to the user
+
+    Return:
+        (list) of fov files with empty json, if none returns empty list
+        raises a warning
+    """
+
+    # retrieve all fovs in bin_file_dir
+    fov_names = remove_file_extensions(io_utils.list_files(bin_file_dir, substrs='.bin'))
+    empty_json_files = []
+
+    # check each json file for size 0
+    for fov in fov_names:
+        fov_path = os.path.join(bin_file_dir, fov + '.json')
+        if os.path.getsize(fov_path) == 0:
+            empty_json_files.append(fov)
+
+    # print a warning to the user when there are empty files
+    if warn and empty_json_files:
```

Comment:
What's the point of the `warn` and `return` conditionals? Are there functions where we don't want the list of empty jsons?